src/NCalc/Domain/EvaluationVisitor.cs

```diff
@@ -699,11 +715,8 @@ public override void Visit(Identifier parameter)
         if (Parameters.ContainsKey(parameter.Name))

```

Comment:
you can replace `ContainsKey` with `TryGetValue` and then use the retrieved value on line 718 and 729

Reply:
Good point! Thanks